x-pack/plugins/security_solution/server/endpoint/services/artifacts/unified_manifes_client.test.ts

```diff
@@ -0,0 +1,253 @@
+      await unifiedManifestClient.updateUnifiedManifests([
+        mockUnifiedManifest(true),
+        mockUnifiedManifest(true),
+      ]);
+      expect(savedObjectsClient.bulkUpdate).toHaveBeenCalledWith(
+        expect.arrayContaining([
+          mockSoClientCallParams({ id: '1234', version: 'abcd' }, true, false),
+          mockSoClientCallParams({ id: '1234', version: 'abcd' }, true, false),
+        ]),
+        { namespace: UNIFIED_MANIFEST_ALL_NAMESPACES }
+      );
+    });
+  });
+  describe('Delete methods', () => {
+    test('can delete unified manifest', async () => {
+      await unifiedManifestClient.deleteUnifiedManifestById('123');
+      expect(savedObjectsClient.bulkDelete).toHaveBeenCalledWith(
+        expect.arrayContaining([{ id: '123', type: ManifestConstants.UNIFIED_SAVED_OBJECT_TYPE }]),
+        { namespace: UNIFIED_MANIFEST_ALL_NAMESPACES }
+      );
+    });
+    test('can delete unified manifests', async () => {
+      await unifiedManifestClient.deleteUnifiedManifestByIds(['123', '456']);
+      expect(savedObjectsClient.bulkDelete).toHaveBeenCalledWith(
+        expect.arrayContaining([
+          mockSoClientCallParams({ id: '123' }, false),
+          mockSoClientCallParams({ id: '456' }, false),
+        ]),
+        { namespace: UNIFIED_MANIFEST_ALL_NAMESPACES }
+      );
+    });
+  });
+  describe('Utility methods', () => {
+    describe('fetchAllUnifiedManifests', () => {
+      const soList = Array.from({ length: 2 }, () => ({
+        updated_at: '2020-01-01T00:00:00.000Z',
+      }));
+
+      const createSOMock = (soResult?: []) => {
+        return {
+          saved_objects: !soResult
+            ? soList.map((soAttributes) => {
+                return {
+                  score: 1,
+                  id: 'so-123',
+                  type: ManifestConstants.UNIFIED_SAVED_OBJECT_TYPE,
+                  version: 'abc',
+                  updated_at: soAttributes.updated_at,
+                  attributes: mockUnifiedManifestAttributes(),
+                  references: [],
+                  sort: ['created_at'],
+                };
+              })
+            : soResult,
+          total: soList.length,
+          per_page: 10,
+          page: 1,
+          pit_id: 'pit-id-1',
+        };
+      };
+
+      beforeEach(() => {
+        savedObjectsClient.find.mockResolvedValueOnce(createSOMock());
+        savedObjectsClient.find.mockResolvedValueOnce(createSOMock());
+        savedObjectsClient.find.mockResolvedValueOnce(createSOMock([]));
+      });
+
+      test('should provide item ids on every iteration', async () => {
+        for await (const items of unifiedManifestClient.fetchAllUnifiedManifests(
+          savedObjectsClient
+        )) {
+          expect(items.map((item) => item.id)).toEqual(['so-123', 'so-123']);
+        }
+
+        expect(savedObjectsClient.find).toHaveBeenCalledTimes(3);
+      });
+
+      test('should use custom options when defined', async () => {
```

Comment:
could you add a test case (or just modify the test parameters) so the `normalizeKuery` function is tested as well?

Reply:
Added with 078703c95fab1e067401685ea46b41aba0bac260